spec/lib/msf/ui/text/dispatcher_shell_spec.rb

```diff
@@ -1,32 +1,264 @@
 require 'spec_helper'
+require 'readline'
 
 RSpec.describe Rex::Ui::Text::DispatcherShell do
-  let(:prompt) { "%undmsf6%clr" }
-  let(:prompt_char) { "%clr>" }
+  let(:prompt) { '%undmsf6%clr' }
+  let(:prompt_char) { '%clr>' }
   let(:subject) do
     dummy_class = Class.new
     dummy_class.include described_class
     dummy_class.new(prompt, prompt_char)
   end
 
+  let(:mock_dispatcher) do
+    dispatcher = double :mock_dispatcher
+    allow(dispatcher).to receive(:tab_complete_helper) do |_current_word, _preceding_words|
+      ['username=']
+    end
+    dispatcher
+  end
+
+  describe '#tab_complete' do
+    let(:dispatcher_stack) do
+      [
+        mock_dispatcher
+      ]
+    end
+
+    before(:each) do
+      allow(subject).to receive(:dispatcher_stack).and_return(dispatcher_stack)
+    end
+
+    [
+      { input: '', expected: nil },
+      { input: '      ', expected: ['      username='] },
```

Comment:
reading the tests and seeing this _not_ try to suggest `password=` confused me a little since it's used in the rest of the tests, also be nice to test completion if more than one option is available